### PR TITLE
Redesign course overview page

### PR DESCRIPTION
**Pull request by terlan98**
| Before | After |
|--------|--------|
| Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-21 at 22 37 52 | Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-05-21 at 22 51 05|

### CODE REVIEW

File: Themis/Views/Exercises/ExerciseSection.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+struct ExerciseSection: View {
+    @EnvironmentObject var courseVM: CourseViewModel
+    
+    var exercises: [Exercise]
+    
+    private let dueDate = ExerciseDateProperty(name: "Submission", dateKeyPath: \.baseExercise.dueDate)
+    private let assessmentDueDate = ExerciseDateProperty(name: "Assessment", dateKeyPath: \.baseExercise.assessmentDueDate)
+    private let releaseDate = ExerciseDateProperty(name: "Release", dateKeyPath: \.baseExercise.releaseDate)
```
Comment (maximiliansoelch) [inline Themis/Views/Exercises/ExerciseSection.swift]: May change the order according to the chronological appearance.

```suggestion
    private let releaseDate = ExerciseDateProperty(name: "Release", dateKeyPath: \.baseExercise.releaseDate)
    private let dueDate = ExerciseDateProperty(name: "Submission", dateKeyPath: \.baseExercise.dueDate)
    private let assessmentDueDate = ExerciseDateProperty(name: "Assessment", dateKeyPath: \.baseExercise.assessmentDueDate)
```

File: Themis/Views/Exams/ExamSection.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+struct ExamSection: View {
+    @EnvironmentObject var courseVM: CourseViewModel
+    
+    var exams: [Exam]
+    var courseID: Int
+    
+    private let startDate = ExamDateProperty(name: "Exam", dateKeyPath: \.startDate)
+    private let assessmentStartDate = ExamDateProperty(name: "Assessment", dateKeyPath: \.startDate)
```
Comment (maximiliansoelch) [inline Themis/Views/Exams/ExamSection.swift]: Are you sure that `\.startDate` should be also used for the `assessmentStartDate`?

File: Themis/Extensions/ExamExtension.swift [hunk 3]
```diff
@@ -16,6 +16,13 @@ extension Exam {
         return false
     }
     
+    var isAssessmentDue: Bool {
+        if let publishResultsDate = self.publishResultsDate {
```
Comment (maximiliansoelch) [inline Themis/Extensions/ExamExtension.swift:20]: How do `assessmentDue` and `publishResultDate` correlate here?

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
I have a few questions and feedback:

Can you align the grey area with the labels on top as you did in your Figma mockup?
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-06-12 at 17 53 31

The last cell has a divider at the bottom:
Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-06-12 at 17 51 33

File: Themis/Extensions/ExamExtension.swift (hunk 3 above)
Comment (terlan98) [reply Themis/Extensions/ExamExtension.swift:20]: The exam model does not have a property indicating when the assessment is due. That's why `publishResultDate` is used instead 🙂

File: Themis/Views/Exams/ExamSection.swift (hunk 2 above)
Comment (terlan98) [reply Themis/Views/Exams/ExamSection.swift]: This is a mistake. Thanks for pointing out.

**Review by maximiliansoelch — APPROVED** [review]
Tested again locally, code looks good! 👍

**Pull request merged**